### ISSUE TITLE
The tabs on the top get in the way for some users

### ISSUE THREAD

**Issue by birtles** [birchill/10ten-ja-reader#688]
e.g.

* https://github.com/birchill/10ten-ja-reader/discussions/675#discussioncomment-1031785
* https://twitter.com/Mirinee/status/1417925030047096842

I think it might be worth having an options to put the tabs on the right-hand side.

**Comment by fishtopher** [comment]
Tabs on bottom would be best I think - and probably less work than rearranging things to make the tabs work in a vertical orientation
Agree with the comment on #675, that bit of space and text between the highlighted word and the dictionary results makes glancing at a definition just that little less smooth than it used to be.

I really like havng the tabs somewhere there though!  It's nice to see which mode you're in, and that there are other options for results!

**Comment by birtles** [comment]
@fishtopher Thank you! That's very good feedback!

I agree that the tabs are helpful for knowing what other results are available and where you are in the set of results.

(For people who find the tabs distracting, this latest release represents a usability regression so I want to address this problem as soon as possible. However, I'm a little exhausted from getting everything else together for this rather big release so it might be a few days before it's ready.)

**Comment by Tomalak** [comment]
I'd like to throw in that tabs as an UI metaphor invite clicking. That's how they work everywhere - and here they don't.

I found myself trying to click on them instinctively more than once, which only leads to a mild frustration as the popup disappears because the mouse has moved off of the highlighted word. Of course it can be learned that _these_ tabs are operated with the Shift key, but it stays a weird deviation from the expectation that tabs in a UI cause.

Maybe the touch version of the UI (where tabs presumably work they way they should) and the non-touch version need to use different approaches how to communicate that there's results in multiple dictionary altogether.

**Comment by birtles** [comment]
Oh, interesting! Yes, I'd like to make the tabs useable by mouse too if possible.

The popup for the Weblioポップアップ英和辞典 is actually clickable. I think that's partly because it works on English text where the word boundaries are easier to detect so you don't need to update the popup as frequently (and hence can keep the same popup even when the mouse moves quite a bit).

We could try to reduce the distance between the mouse and the popup and make it clickable but I suspect you'd end up moving your mouse diagonally and the popup would update.

I want to add a "sticky" mode where you press, e.g. <kbd>s</kbd> and then the popup stays and probably switches to the touch version with the larger tabs. That still doesn't address the issue of user expectations, however.

**Comment by Tomalak** [comment]
I think it's also worth noting that the mouse pointer position itself has merit:

As the user reads the text and moves the mouse on a word they'd like to look up, their eyes move off the word and onto the popup. When they  now move away the mouse pointer from the word in order to operate UI features in the popup, the previous, "visually bookmarked" position in the text is lost. When they move the mouse out of the popup again, they likely end up on an random unrelated word and a new popup appears.

**Comment by birtles** [comment]
I made a start on this today:

![image](https://user-images.githubusercontent.com/1232595/126860393-b9fda7ad-a383-401a-a9af-f2fb401d8c5b.png)

I also noticed that Yomichan makes the popup interactive so maybe it's not as impractical as I first thought.

> I think it's also worth noting that the mouse pointer position itself has merit:

That's a good point. For users with the default "highlight text" setting enabled, hopefully that helps guide you back to your place? It wouldn't, however, help with the fact that you'll probably stick your mouse on somewhere unrelated.

**Comment by birtles** [comment]
I won't get to look at this again until Monday so I've pushed the WIP patch to the [`tabs-on-side`](https://github.com/birchill/10ten-ja-reader/tree/tabs-on-side) branch.

**Comment by Tomalak** [comment]
"Tabs on the side" is definitely better than on top.

In a touch screen scenario, switching tabs does not require obstructing the popup with your hand, which is a usability plus. On that note it would be beneficial to offer tabs on the left or right, to accommodate left-handed and right-handed people, or different ways of holding the device.

For non-touch devices, tabs on the side is also an improvement IMO, but personally, I would actually prefer a way to disable them altogether. I'm used to the workflow of flipping through other results with the Shift key if I need them, the visual indication that there are matches in other dictionaries does not add all that much for me. I either already know (i.e. on a Kanji) or strongly suspect it (i.e. on a word that is very probably a name). In both cases it takes me minimal effort to check.

**Comment by alexulter** [comment]
I would suggest making "interactive popup" optional too. It was actually one of the reasons I personally preferred Rikaichan over Yomichan, like it's "lightweight" and fast way to just hover mouse above words and have popup hint you gently about their meaning/reading.

**Comment by Tomalak** [comment]
...maybe along the lines of this so users don't have to dive into the preferences to change the layout

![image](https://user-images.githubusercontent.com/28300/126871399-47c9dd58-d778-41a5-baba-ed56a4ff56d1.png)

**Comment by alexulter** [comment]
@Tomalak But it's something one doesn't change often. It would be perfectly ok to change it once in options menu and then forget about it since then.

**Comment by birtles** [comment]
Some excellent feedback here. Thanks @Tomalak and @alexulter! I have some idea about how to address those different points but let me try it out first.

**Comment by Tomalak** [comment]
Oh and other thing I noticed:

The icons for "Words" and "Kanji" seem too fiddly and crowded. I think their border takes away too much of the icon real estate (plus, the person icon has no border either). Yeah,. it's not technically a border with the "open book" icon, but it does resemble one:

![image](https://user-images.githubusercontent.com/28300/126959491-8298542d-86fd-4f04-a8a7-e4a23de32d5e.png)

I'd stay with a stylized, borderless 字 for the "Kanji" icon, and maybe something more abstract for the "Words" dictionary, for example along the lines of `book-bookmark` or maybe even `magnifying-glass` from Font-Awesome.

| ![image](https://user-images.githubusercontent.com/28300/126963402-c46d8b1a-fb25-4333-a15d-e5461c5697ee.png) | ![image](https://user-images.githubusercontent.com/28300/126963326-82fe3112-7bda-42e7-9db2-b54c5d62c08f.png) |
| --- | --- |

**Comment by birtles** [comment]
> The icons for "Words" and "Kanji" seem too fiddly and crowded. I think their border takes away too much of the icon real estate (plus, the person icon has no border either). Yeah, it's not technically a border with the "open book" icon, but it does resemble one:

Oh I spent hours making these icons and tweaking them to get them pixel perfect 😅

> Or one could go all-in, and use 語 or 言 for "Words", 字 for "Kanji" and 人 for "People" / 名 for "Names". Arguably, any learner of Japanese will recognize these early-on, and they would implicitly not require the text label that goes missing when moving the tabs to the side.

Right, I considered doing that but when you're using the Japanese version of the add-on with tabs on top you'd get:

`[語] 単語 | [字] 漢字 | [人] 人名`

which seems redundant and hard to read. I put the `字` in a box simply so that it didn't appear to be part of the label. Here's how it currently looks when you're using Japanese Firefox:

![image](https://user-images.githubusercontent.com/1232595/126985100-ca789877-0d44-4373-ba67-dbf6bbe0e61e.png)

**Comment by Tomalak** [comment]
> but when you're using the Japanese version of the add-on with tabs on top you'd get:

...that's why "tabs on the side and no label text" would lend itself to the idea. :)

**Comment by birtles** [comment]
> > but when you're using the Japanese version of the add-on with tabs on top you'd get:
> 
> ...that's why "tabs on the side and no label text" would lend itself to the idea. :)

Fair enough. I'm happy to look into separate icons for the vertical tab states but I'd rather do that as a separate issue once the vertical tabs have landed (and they're getting close -- just working on the options page).

**Comment by Tomalak** [comment]
Yeah I did not intend to muddy up the issue at hand, feel free to separate things here.

**Comment by birtles** [comment]
I've merged the work here into the main branch now so this should be included in the next release.

It looks something like this:

https://user-images.githubusercontent.com/1232595/127097223-f432f797-c8d4-4f45-aecb-8851b46e1880.mp4

The other idea I had was to add a settings icon when in touch mode. It looks like this:

![image](https://user-images.githubusercontent.com/1232595/127097518-8bdd0630-d872-4730-9628-322e9400ab54.png)

Or in vertical mode:

![image](https://user-images.githubusercontent.com/1232595/127097558-841263d8-b747-4ae0-89a8-b3b3b5535f87.png)

Tapping the settings icon causes the settings page to be immediately opened (in a new tab).

I'm not entirely sure if it's a good idea or not so I've pushed it to the [`settings-icon`](https://github.com/birchill/10ten-ja-reader/tree/settings-icon) branch for now.
